api/src/main/java/org/cardanofoundation/rosetta/api/block/service/LedgerBlockServiceImpl.java

```diff
@@ -75,14 +83,34 @@ public Optional<Block> findBlock(Long blockNumber, String blockHash) {
     }
   }
 
+  @Override
+  public Optional<BlockIdentifierExtended> findBlockIdentifier(Long blockNumber, String blockHash) {
+    log.debug("Query blockNumber: {} , blockHash: {}", blockNumber, blockHash);
+    if (blockHash == null && blockNumber != null) {
+      return blockRepository.findBlockIdentifierByNumber(blockNumber)
+          .map(blockMapper::mapToBlockIdentifierExtended);
+    } else if (blockHash != null && blockNumber == null) {
+      return blockRepository.findBlockIdentifierByHash(blockHash)
+          .map(blockMapper::mapToBlockIdentifierExtended);
+    } else {
+      return blockRepository
+          .findBlockIdentifierByNumberAndHash(blockNumber, blockHash)
+          .map(blockMapper::mapToBlockIdentifierExtended);
+    }
+  }
+
   @NotNull
   private Block toModelFrom(BlockEntity blockEntity) {
     Block model = blockMapper.mapToBlock(blockEntity);
-    model.getTransactions().forEach(this::populateTransaction);
+    ProtocolParams pps = protocolParamService.findProtocolParametersFromIndexer();
+    List<BlockTx> transactions = model.getTransactions();
+    Entities fetched = findByTxHash(transactions);
+    transactions.forEach(tx -> populateTransaction(tx, pps, fetched));
     return model;
   }
 
   @Override
+  @Transactional(readOnly = true)
```

Comment:
Here we need to set a propagation level. Also, this could help in resolving excessive connection problems:
https://stackoverflow.com/questions/52177887/multi-threading-spring-transaction

Reply:
We already have propagation level in the beginning of  the class
@Transactional(readOnly = true, propagation = Propagation.SUPPORTS)
Removed